doc/getting-started.md

```diff
@@ -0,0 +1,344 @@
+
+In a terminal, run :
+```bash
+# Headless node
+$ dogecoind -printtoconsole
+
+# GUI
+$ dogecoin-qt
+```
+
+*Use `--help` to see all available options for each executables.*
+
+Your node should now be running, starting a *syncing process* by downloading the entire blockchain from other nodes. This operation will take hours, days to complete... but your now part of the dogecoin network !
+
+> **Warning :** The rest of the guide will assume the use of an headless node. The RPC server is not exposed with `dogecoin-qt` until you activate the `-server` option.
+
+## Introduction to JSON-RPC API
+
+Dogecoin Core expose a JSON-RPC interface. With it, you can request information about blocks and transactions, send transactions to the networks, manage your wallet and so on...
+
+To interact with the interface, it relies on HTTP requests. Using urls like `http://user:password@127.0.0.1:22555/` and sending commands through json format is feasible.
+
+Some tools like libraries can be built on top of it. `dogecoin-cli` is the default shell client provided with Dogecoin Core to use this API.
+
+The RPC API documentation is available from [bitcoin](https://developer.bitcoin.org/reference/rpc/).  
+To have an overview of all commands, see also :
+
+```bash
+# List all commands
+$ dogecoin-cli help
+
+# Get help for a specific command
+$ dogecoin-cli help COMMAND
+```
+> **Tips** : Abuse of the help menus is allowed ! They can answer to multiple questions.
+
+### Create a wallet
+
+To receive or recover coins, you will need a wallet address and a private key. The *wallet address* is a shareable information to receive dogecoins from all over the world, while the *private key* is a sensitive information who give the ownership on coins associated to a public address.
+
+By default, you already have a wallet address and a private key available.
+
+List it using `getaddressesbyaccount` :
+
+```bash
+$ dogecoin-cli getaddressesbyaccount ""
+[
+  "DBZc6qeBuo7g8La7L3Lhaf39MsJSgbzpMX"
+]
+```
+
+Use `getnewaddress` to generate a new wallet address :
+```bash
+$ dogecoin-cli getnewaddress
+DC13A8qUgAu5rTWweb44TwXZNQbxDYhyo2
+```
+
+And `dumpprivkey` to find the corresponding private key.
+```bash
+$ dogecoin-cli dumprivkey DC13A8qUgAu5rTWweb44TwXZNQbxDYhyo2
+QSqPGRwxeqv8KNEU85jw128g7rt5ognjqaSvjmeouuFAjJ8Bq3ZR
+```
+
+> **Tips :** Dogecoin wallets addresses mainly start with the letter `D` and private keys with `Q`.
+
+You now have two wallet addresses to share to anyone who want to give you some dogecoins !
+
+### Send a transaction
+
+Dogecoin Core follow the [Unspent Transaction Output (UTXO)](https://en.wikipedia.org/wiki/Unspent_transaction_output) model to manage balances. To own some dogecoins is like having a list of unspent transactions. To spend them, you have to compose a new transaction with received unspent transaction.
+
+Four commands to execute to be able to tip : `listunspent`, `createrawtransaction`, `signrawtransaction` and `sendrawtransaction`.
```

Comment:
Why not just `sendtoaddress`?

Reply:
Mainly because I followed mastering bitcoin, and they were using this flow. I don't know personally used this command, also I was concern about giving granularity to choose transaction/wallets.

I didn't explore the command enough, but I was also worried to have the same behavior I experienced on dogechain, who pick UTXO from multiple wallet and send money. It gives more the low-level information.

Reply:
Do you need that to get started? Or is that the advanced course?

Reply:
Depend on who you are, again it's also targeting devs who may want to create application, and for this guy I supposed it's useful and a basic information.

Reply:
What type of application?

Reply:
Whatever it is, in this case something which implement transactions. Let's say a video game for example.

Reply:
My question is more what does it implement transactions for? What does it do? 

Does it pay out pre-funded DOGE to players? Then sure, but you can probably use `sendtoaddress` or worst case `fundrawtransaction` and suddenly you don't have to write your own coin selection algorithm, unless you think the one in the wallet is bad (but if you have an opinion about that, then you likely don't need to read this guide.) 

Does it allow players to do micro purchases / trades (think steam's marketplace)? Then you'll probably want payment channels (for speed) and your own non-custodial wallet software inside your game (to not become a money transmitter / bank) so you won't use the wallet for transactions.

I really don't see much of a use-case here except for wanting to create your own coin selection algorithm. But that's hardly "getting started" material. I also strongly doubt whether we should target developers in this guide as I mentioned above: the recommendations for developers to use Dogecoin Core to interface with the blockchain are much more involved and require much more thought than we're giving to this right now. I've been helping devs bootstrap crypto applications for over 7 years now and the things you'll need to run your application, even if it's just for accepting simple payments, are much greater than what we're describing here. 

I'm willing to work on another set of documents for developers with you if you aspire to cover that reader segment too, but if we really want to use Dogecoin Core for this, we'll have to do a bunch of enhancements to the actual software too. I do think it's a great goal to have though.

Reply:
I think we should consider targeting developer rather than simple users, at first. Because we may expect developers to work on user documentation, while the opposite isn't possible. In my mind, it lacks devs around the project so it's why I try to target potential developers, and in any case it uses CLI mainly, it reduces drastically potential readers.

You're definitively more skilled than me to know what is needed for devs to build an application, it's not enough, but I hope it's more resources than it currently is. I see it more as a starting point to build more complete documentation. In this guide, I'm translating "Mastering Bitcoin" content into the documentation. That's why you see the transaction process split, the transaction decoded in two steps. Because they are doing things in that way.

Reply:
It's good to add `sendtoaddress`, but I think it can be good to keep the process split to. For educational purpose it may be great, and it can give idea to people, where you can build your transaction and sign it later.

Showing commands split, and then show a transaction with `sendtoaddress`, to show there is a simple way to do it.

Reply:
> In my mind, it lacks devs around the project so it's why I try to target potential developers

In that case, I propose to postpone this. Your description of this PR was to target users, not so much devs. Creating developer documentation is very different from "getting started" and should be much more about integration patterns and how to realize them. This is not what this document does at all. 

>  I'm translating "Mastering Bitcoin" content into the documentation

That's published under CC4 ShareAlike which is not compatible with MIT under which this document would ideally be published here. So now we may have an issue.

Reply:
It reuses the process, not the content. Nothing is copied, and used commands are done with new wallet/transaction/whatever. It shouldn't have any issue regarding copyright.

Difficult to be accurate about who it's targeting, the guy who will use Dogecoin through CLI is someone with programming skills, so a potential developer. Idk what is a typical "Dogecoin core" user.

I agree it would be great to have a well-delimited developer doc and a user doc, for now what I'm suggesting is a getting started guide for people with programming skills to be able to play with Dogecoin core, so simultaneously potential devs. It's to help people to jump in with the core to expect in the future to go in this direction and expect to get a more documented software.
+ potentially guys from the foundation were speaking about "dogepedia", it may complete this gap for more common users. Perhaps one day they will help on docs too.

Personally, I won't work on dev docs. First because I do not have the experience/skill + I would like to finish my open PRs and I will go away.

Reply:
> Difficult to be accurate about who it's targeting

Okay, so let's do a simple `getbalance` and `sendtoaddress`, and the more advanced manual coin selection both? 

> I would like to finish my open PRs and I will go away.

😕 I guess that means I'll get to take over maintenance of the docs you're creating...

Reply:
The entire documentation in overall need to be maintained and to evolve, like you said more content is needed with a developer section with detailed instruction, API documentation, security recommendation, contribution guides, whatever. Idk what would come from bitcoin and be specific to Dogecoin, but there are many areas of improvement for the documentation, and in any case its endless work through the software lifecycle.

We reworked installation instruction, may add this getting started section + docker. The idea was to facilitate developer path when they start with Dogecoin Core, to help to involve more people inside the ecosystem. I found it rude to start when I first come here, I hope what we are doing is useful to make it easier.

I added content for `getbalance` and `sendtoaddress`.